libfreerdp/codec/h264_mediacodec.c

```diff
@@ -0,0 +1,516 @@
+                pYUVData[2] = outputBuffer + iStride[0] * sys->height + iStride[1] * ((sys->height + 1) / 2);
+                break;
+            }
+            else if (outputBufferId ==  AMEDIACODEC_INFO_OUTPUT_FORMAT_CHANGED)
+            {
+                AMediaFormat* outputFormat;
+                outputFormat = sys->fnAMediaCodec_getOutputFormat(sys->decoder);
+                if (outputFormat == NULL)
+                {
+                    WLog_Print(h264->log, WLOG_ERROR, "AMediaCodec_getOutputFormat failed");
+                    return -1;
+                }
+                set_mediacodec_format(h264, &sys->outputFormat, outputFormat);
+
+                media_format = sys->fnAMediaFormat_toString(sys->outputFormat);
+                if (media_format == NULL)
+                {
+                    WLog_Print(h264->log, WLOG_ERROR, "AMediaFormat_toString failed");
+                    return -1;
+                }
+            }
+            else if (outputBufferId ==  AMEDIACODEC_INFO_TRY_AGAIN_LATER)
+            {
+                WLog_Print(h264->log, WLOG_WARN, "AMediaCodec_dequeueOutputBuffer need to try again later");
+                // TODO: sleep?
+            }
+            else if (outputBufferId == AMEDIACODEC_INFO_OUTPUT_BUFFERS_CHANGED)
+            {
+                WLog_Print(h264->log, WLOG_WARN, "AMediaCodec_dequeueOutputBuffer returned deprecated value AMEDIACODEC_INFO_OUTPUT_BUFFERS_CHANGED, ignoring");
+            }
+            else
+            {
+                WLog_Print(h264->log, WLOG_ERROR, "AMediaCodec_dequeueOutputBuffer returned unknown value [%d]", outputBufferId);
+                return -1;
+            }
+        }
+
+        break;
+    }
+
+    return 1;
+}
+
+static void mediacodec_uninit(H264_CONTEXT* h264)
+{
+    H264_CONTEXT_MEDIACODEC* sys = (H264_CONTEXT_MEDIACODEC*)h264->pSystemData;
+    media_status_t status = AMEDIA_OK;
+
+    WLog_Print(h264->log, WLOG_INFO, "Uninitializing MediaCodec");
+
+	if (!sys)
+		return;
+
+
+    if (sys->decoder != NULL)
+    {
+        release_current_outputbuffer(h264);
+        status = sys->fnAMediaCodec_stop(sys->decoder);
+        if (status != AMEDIA_OK)
+        {
+            WLog_Print(h264->log, WLOG_ERROR, "Error AMediaCodec_stop %d", status);
+        }
+
+        status = sys->fnAMediaCodec_delete(sys->decoder);
+        if (status != AMEDIA_OK)
+        {
+            WLog_Print(h264->log, WLOG_ERROR, "Error AMediaCodec_delete %d", status);
+        }
+
+        sys->decoder = NULL;
+    }
+
+    set_mediacodec_format(h264, &sys->inputFormat, NULL);
+    set_mediacodec_format(h264, &sys->outputFormat, NULL);
+
+    unload_libmediandk(h264);
+
+    free(sys);
+	h264->pSystemData = NULL;
+}
+
+static BOOL mediacodec_init(H264_CONTEXT* h264)
+{
+    H264_CONTEXT_MEDIACODEC* sys;
+    media_status_t status;
+    const char* media_format;
+    char* codec_name;
+    AMediaFormat* inputFormat, *outputFormat;
+
```

Comment:
Please add `WINPR_ASSERT(h264)` (and similar checks for the functions called).
Not done in (all) existing code, but we´d like to get a cleaner base in the future.